conans/model/build_info.py

```diff
@@ -139,12 +158,12 @@ def __init__(self, root_folder):
         self.resdirs.append(DEFAULT_RES)
         self.builddirs.append(DEFAULT_BUILD)
         self.frameworkdirs.append(DEFAULT_FRAMEWORK)
+        self.components = DefaultOrderedDict(lambda: Component(self.rootpath))
```

Comment:
We can avoid the new Component class if there is a way of setting the rootpath of the _CppInfo object created here, but I don't know if there is an alternative about this

Reply:
I wonder if we need to add the `rootpath` to the components...

Reply:
I think it is not needed but it was done this way to reuse the _CppInfo class and keep the same structure. At this point, we might also consider adding full Component class, but I wanted to add the minimal code possible to this PR

Reply:
We probably don't need the rootpath in the components.

Reply:
Yes, at the moment we need the rootpath to compose, it is the simpler way. In the future, we will see depending on the usage patterns in generators.